awx/settings/development.py

```diff
@@ -105,8 +105,11 @@
 AWX_DISABLE_TASK_MANAGERS = False
 # ======================!!!!!!! FOR DEVELOPMENT ONLY !!!!!!!=================================
 
-if 'sqlite3' not in DATABASES['default']['ENGINE']:  # noqa
-    DATABASES['default'].setdefault('OPTIONS', dict()).setdefault('application_name', f'{CLUSTER_HOST_ID}-{os.getpid()}-{" ".join(sys.argv)}'[:63])  # noqa
+from .application_name import set_application_name
+
+set_application_name(DATABASES, CLUSTER_HOST_ID)
+
+del set_application_name
```

Comment:
This module is what will result when you import Django settings, which you get doing

```python
from django.conf import settings
```

The settings are pretty special compared to the rest of the code, and comes _way earlier_ in import order than most stuff. So you really can't from other places in your app, because you will very quickly get circular imports. In this case, however, I have a static method that I want to share around. You can define methods / classes / logic in your settings, but you should try to avoid this as much as possible. My solution is that I introduce a new module inside of the settings folder that contains static methods. The ordinary settings use one of those static methods, but I'm not trying to vendor that static method inside of the settings.

In other words, I don't want someone to be able to do `settings.set_application_name()`. It's not that it would be invalid, but I'm doing the imports a different way, because my design preference is to not define methods inside of settings. Ideally, I want settings to contain _static_ content.

So deleting `set_application_name` removes that method from the local variables in settings, but the result of running it remains, in that it has altered the `DATABASES` local variable. I just want to be very careful about what I _add to_ settings, so I do this thing, but I delete the python resources I used in order to do it.